apps/livreGUI/Controller.cpp

```diff
@@ -149,8 +149,12 @@ struct Controller::Impl
         */
         while( _continuePolling )
         {
-            ScopedLock lock( _subscriberMutex );
-            BOOST_FOREACH( const URISubscriberPair& pair, _subscriberMap )
+            SubscriberMap subscribers;
+            {
+                ScopedLock lock( _subscriberMutex );
+                subscribers = _subscriberMap;
+            }
+            BOOST_FOREACH( const URISubscriberPair& pair, subscribers )
             {
                 zeq::Subscriber* subscriber = pair.second;
                 subscriber->receive( 100 );
```

Comment:
Don't fully understand, but is solves deadlocks when registering/deregistering subscribers while message are received.